src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java

```diff
@@ -196,10 +195,50 @@ public void deleteAttachment(String requestId, String companyNumber, String obje
                 () -> new AttachmentNotFoundException(String.format(ATTACHMENT_NOT_FOUND_MESSAGE, attachmentId))
         );
 
-        attachments.remove(attachment);
+        Map<String, Object> logMap = buildLogMap(companyNumber, objectionId);
+        boolean successfulDelete = deleteFromS3(requestId, attachmentId, logMap);
+
+        if(successfulDelete) {
+            attachments.remove(attachment);
+
+            objection.setAttachments(attachments);
+
+            objectionRepository.save(objection);
+        } else {
+            throw new ServiceException(UNABLE_TO_DELETE_ATTACHMENT);
+        }
+    }
 
-        objection.setAttachments(attachments);
+    private boolean deleteFromS3(String requestId, String attachmentId, Map<String, Object>  logMap) {
+        try {
+            FileTransferApiClientResponse response = fileTransferApiClient.delete(requestId, attachmentId);
```

Comment:
rather than return lots of true/false it would be better to throw an exception if unsuccessful

Reply:
Reworked this.